library/Faktory/Worker.hs

```diff
@@ -67,18 +67,16 @@ runWorker settings queue f = do
 processorLoop :: FromJSON arg => Client -> Queue -> (arg -> IO ()) -> IO ()
 processorLoop client queue f = do
   let
-    processAndAck job = do
-      f $ jobArg job
-      ackJob client job
+    processAndAck job =
+      tryAny (f $ jobArg job) >>= \case
+        Left ex -> failJob client job $ T.pack $ show ex
+        Right _ -> ackJob client job
 
   mJob <- fetchJob client queue
 
   case mJob of
     Just job ->
-      processAndAck job `catches`
-        [ Handler $ \(ex :: WorkerHalt) -> throw ex
-        , Handler $ \(ex :: SomeException) -> failJob client job $ T.pack $ show ex
-        ]
+      processAndAck job `catch` \(ex :: WorkerHalt) -> throw ex
```

Comment:
Hmm, actually I think this can stay the way it is. A failed ack would eventually result in another `fetchJob` call, and `tryAny` prevents `WorkerHalt` from being caught here.

Reply:
Yeah, I was typing up this exact question. I think this should be put back the way it was.